presto-main/src/main/java/io/prestosql/spiller/AesSpillCipher.java

```diff
@@ -0,0 +1,193 @@
+    private static final String CIPHER_NAME = "AES/CBC/PKCS5Padding";
+    private static final int KEY_BITS = 256;
+
+    private SecretKey key;
+    private final int ivBytes;
+
+    AesSpillCipher()
+    {
+        this.key = generateNewSecretKey();
+        this.ivBytes = createEncryptCipher().getIV().length;
+    }
+
+    @Override
+    public ByteBuffer decrypt(ByteBuffer encryptedData)
+    {
+        int length = encryptedData.remaining();
+        if (length < ivBytes) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Encrypted data size is too small: %s. It must be at least the size of the initialization vector: %s.", length, ivBytes));
+        }
+        byte[] iv = new byte[ivBytes];
+        encryptedData.get(iv);
+        Cipher cipher = createDecryptCipher(new IvParameterSpec(iv));
+        ByteBuffer output = ByteBuffer.allocate(cipher.getOutputSize(encryptedData.remaining()));
+        try {
+            cipher.doFinal(encryptedData, output);
+            output.flip();
+            return output;
+        }
+        catch (GeneralSecurityException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Cannot decrypt previously encrypted data: " + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public byte[] decrypt(byte[] encryptedData)
+    {
+        return unwrapOrCopyToByteArray(decrypt(ByteBuffer.wrap(encryptedData)));
+    }
+
+    @Override
+    public ByteBuffer encrypt(ByteBuffer data)
+    {
+        Cipher cipher = createEncryptCipher();
+        byte[] iv = cipher.getIV();
+        ByteBuffer output = ByteBuffer
+                .allocate(iv.length + cipher.getOutputSize(data.remaining()))
+                .put(iv);
+        try {
+            cipher.doFinal(data, output);
+            output.flip();
+            return output;
+        }
+        catch (GeneralSecurityException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Failed to encrypt data: " + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public byte[] encrypt(byte[] data)
+    {
+        return unwrapOrCopyToByteArray(encrypt(ByteBuffer.wrap(data)));
+    }
+
+    /**
+     * Setting the {@link AesSpillCipher#key} to null allows the key to be reclaimed by GC at the time of
+     * destruction, even if the {@link AesSpillCipher} itself is still reachable for longer. When the key
+     * is null, subsequent encrypt / decrypt operations will fail and can prevent accidental use beyond the
+     * intended lifespan of the {@link AesSpillCipher}.
+     */
+    @Override
+    public void destroy()
+    {
+        this.key = null;
+    }
+
+    @Override
+    public boolean isDestroyed()
+    {
+        return key == null;
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this).add("isDestroyed", isDestroyed()).toString();
+    }
+
+    private SecretKey verifyKeyNotDestroyed()
+    {
+        SecretKey key = this.key;
```

Comment:
The code doesn't try to be thread safe and (hopefully) doesn't need to be. No need for a local variable.